src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,244 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get all cars by drivers id: " + driverId, e);
+        }
+        addDriversForCars(cars);
+        return cars;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setManufacturer(getManufacturer(resultSet.getObject("manufacturer_id", Long.class)));
+        car.setId(resultSet.getObject("car_id", Long.class));
+        return car;
+    }
+
+    private Manufacturer getManufacturer(Long id) {
+        String query = "SELECT * FROM manufacturers WHERE id = "
+                + id + " AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement statement = connection.createStatement()) {
+            ResultSet resultSet = statement.executeQuery(query);
+            Manufacturer manufacturer = null;
+            if (resultSet.next()) {
+                manufacturer = setManufacturer(resultSet);
+            }
+            return manufacturer;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get manufacturer by id " + id + " ",
+                    throwable);
+        }
+    }
+
+    private Manufacturer setManufacturer(ResultSet resultSet) throws SQLException {
+        Long newId = resultSet.getObject("id", Long.class);
+        String name = resultSet.getString("name");
+        String country = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(name, country);
+        manufacturer.setId(newId);
+        return manufacturer;
+    }
+
+    private Driver getDriverFromResultSet(ResultSet resultSet) throws SQLException {
+        Driver driver = new Driver(resultSet.getString("name"),
+                resultSet.getString("licence"));
+        driver.setId(resultSet.getObject("id", Long.class));
+        return driver;
+    }
+
+    private List<Driver> getDrivers(Long id) {
+        String query = "SELECT * FROM drivers JOIN cars_drivers "
+                + "ON cars_drivers.driver_id = drivers.id "
+                + "WHERE cars_drivers.car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversStatement =
+                         connection.prepareStatement(query)) {
+            getDriversStatement.setLong(1, id);
+            ResultSet resultSet = getDriversStatement.executeQuery();
+            List<Driver> driverList = new ArrayList<>();
+            while (resultSet.next()) {
+                driverList.add(getDriverFromResultSet(resultSet));
+            }
+            return driverList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get drivers for car id: " + id, e);
+        }
+    }
+
+    private void addDriversForCars(List<Car> cars) {
+        if (cars.size() > 0) {
```

Comment:
I think it would be better to add a null check here. Because if the size of the list is zero, we won't even get into this loop. But if it is null, we can get NPE.